### PR TITLE
Add conversion from Value into Emscripten's val

### PR DESCRIPTION
**Pull request by emaxx-google**
Implement helper function that transforms a Value object into
the Emscripten Embind's emscripten::val object.

This will be used in follow-ups in order to send messages to
JavaScript in Emscripten/WebAssembly builds, with the messages
prepared by a toolchain-independent Value class, contributing to
the goal of making most of the //common/cpp library be
toolchain-independent (as tracked by #185).

### CODE REVIEW

File: common/cpp/src/google_smart_card_common/value_emscripten_val_conversion_unittest.cc [hunk 1]
```diff
@@ -0,0 +1,205 @@
...
+    ASSERT_TRUE(converted.isString());
+    EXPECT_EQ(converted.as<std::string>(), kFoo);
+  }
+}
+
+TEST(ValueEmscriptenValConversion, BinaryValue) {
+  {
+    const emscripten::val converted =
+        ConvertValueToEmscriptenVal(Value(Value::Type::kBinary));
+    ASSERT_TRUE(converted.instanceof (emscripten::val::global("ArrayBuffer")));
```
Comment (emaxx-google) [inline common/cpp/src/google_smart_card_common/value_emscripten_val_conversion_unittest.cc:126]: Hmm, this looks like a clang-format bug. This `instanceof()` is a regular C++ class method, there shouldn't be a space after it. I guess we can keep it as it is. The alternative is to remove the space and surround this code with `// clang-format off`.

**Review by Fabian-Sommer — APPROVED** [review]
LGTM

File: common/cpp/src/google_smart_card_common/value_emscripten_val_conversion_unittest.cc (hunk 1 above)
Comment (MahmoudAGawad) [reply common/cpp/src/google_smart_card_common/value_emscripten_val_conversion_unittest.cc:126]: I am leaning towards disabling clang-format in this line. It feels like a special operator while in fact it is a class method. Or maybe clang has an rational explanation for this? (I couldn't find any)

**Review by MahmoudAGawad — APPROVED** [review]
LGTM.

**Pull request merged**